### PR TITLE
Add AUDITLOG_TRUNCATE_CHANGES_DISPLAY and AUDITLOG_TRUNCATE_LIMIT

### PR DESCRIPTION
**Pull request by hoangquochung1110**
To configure how many characters will be truncated or disable it

Resovle #633

### CODE REVIEW

File: auditlog_tests/test_settings.py [hunk 1]
```diff
@@ -26,15 +26,12 @@
...
-
 DATABASES = {
     "default": {
         "ENGINE": "django.db.backends.postgresql",
-        "NAME": os.getenv(
-            "TEST_DB_NAME", "auditlog" + os.environ.get("TOX_PARALLEL_ENV", "")
-        ),
-        "USER": os.getenv("TEST_DB_USER", "postgres"),
-        "PASSWORD": os.getenv("TEST_DB_PASS", ""),
+        "NAME": "alana",
```
Comment (hramezani) [inline auditlog_tests/test_settings.py:32]: Probably these are your local change and not needed

File: auditlog/conf.py [hunk 2]
```diff
@@ -45,3 +45,11 @@
...
     settings, "AUDITLOG_DISABLE_REMOTE_ADDR", False
 )
+
+# Enable changes_display_dict truncator
+settings.AUDITLOG_TRUNCATE_CHANGES_DISPLAY = getattr(
+    settings, "AUDITLOG_TRUNCATE_CHANGES_DISPLAY", True
+)
+
+# Number of characters at which changes_display_dict property should be shown
+settings.AUDITLOG_TRUNCATE_LIMIT = getattr(settings, "AUDITLOG_TRUNCATE_LIMIT", 140)
```
Comment (hramezani) [inline auditlog/conf.py:55]: I think we can handle this with only one setting like `AUDITLOG_CHANGE_DISPLAY_TRUNCATE_LENGTH`.

- If the value is `0` -> display nothing
- if the values is > `0` -> truncate
- if the values < `0` -> display all the string

it has to be documented as well.

What do you think?

File: auditlog/models.py [hunk 3]
```diff
@@ -506,9 +507,11 @@ def changes_display_dict(self):
...
                         value = self._get_changes_display_for_fk_field(field, value)
 
-                    # check if length is longer than 140 and truncate with ellipsis
-                    if len(value) > 140:
-                        value = f"{value[:140]}..."
+                    if (
+                        settings.AUDITLOG_TRUNCATE_CHANGES_DISPLAY
+                        and len(value) > settings.AUDITLOG_TRUNCATE_LIMIT
+                    ):
+                        value = truncatechars(value, settings.AUDITLOG_TRUNCATE_LIMIT)
```
Comment (hramezani) [inline auditlog/models.py:514]: as the logic is simple, I would prefer to keep it here instead of `Truncator` class

**Comment by hramezani** [comment]
Thanks @hoangquochung1110 for the PR, I left a couple of comments.

Please add a changelog

**Comment by hoangquochung1110** [comment]
Appreciate your time and review
I've updated the PR accordingly
Also help me check the usage and changelog as english is not my mother tongue.
Thank you @hramezani

File: docs/source/usage.rst [hunk 4]
```diff
@@ -305,6 +305,14 @@ If the value is `None`, the default getter will be used.
...
 .. versionadded:: 3.0.0
 
+**AUDITLOG_CHANGE_DISPLAY_TRUNCATE_LENGTH**
+
+Number of characters at which strings in `changes_display_dict` property should be truncated if length exceeded
+
+Set to 140 by default.
+
+You can use this settings to preserve the strings by set it to a negative number.
+
```
Comment (hramezani) [inline docs/source/usage.rst:315]:
```suggestion
This configuration variable defines the truncation behavior for strings in `changes_display_dict`, with a default value of `140` characters.

0: The entire string is truncated, resulting in an empty output.
Positive values (e.g., 5): Truncates the string, keeping only the specified number of characters followed by an ellipsis (...) after the limit.
Negative values: No truncation occurs, and the full string is displayed.

.. versionadded:: 3.1.0

```

**Comment by hramezani** [comment]
Thanks @hoangquochung1110 for the update.

LGTM, I only left a suggestion for the new config description.

**Review by hramezani — APPROVED** [review]

**Comment by hramezani** [comment]
Thanks @hoangquochung1110

**Comment by hoangquochung1110** [comment]
Thank you for your approval @hramezani

**Pull request merged**